deployments/auxiliary/cloudformation/panther-bi-iam-permissions.yml

```
@@ -0,0 +1,112 @@
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+AWSTemplateFormatVersion: 2010-09-09
+Description: IAM permissions needed to connect to Athena through third party tools.
+
+Resources:
+  Athena:
+    Type: AWS::IAM::User
+    Properties:
+      UserName: Athena_analytics_readonly
+
+  AthenaGroupMembership:
+    Type: AWS::IAM::UserToGroupAddition
+    Properties:
+      GroupName: !Ref AthenaGroup
+      Users:
+        - !Ref Athena
+
+  AthenaGroup:
+    Type: AWS::IAM::Group
+    Properties:
+      GroupName: AthenaAnalytics
+
+  AthenaGroupPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: ReadPantherProcessedData
+      Groups:
+        - !Ref AthenaGroup
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - athena:BatchGetQueryExecution
+              - athena:CancelQueryExecution
+              - athena:Get*
+              - athena:List*
+              - athena:RunQuery
+              - athena:StartQueryExecution
+              - athena:StopQueryExecution
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - glue:BatchGetPartition
+              - glue:Get*
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - s3:AbortMultipartUpload
+              - s3:GetBucketLocation
+              - s3:GetObject
+              - s3:ListBucket
+              - s3:ListBucketMultipartUploads
+              - s3:ListMultipartUploadParts
+              - s3:PutObject
+            Resource: arn:aws:s3:::panther-bootstrap-athenaresults-*
+          - Effect: Allow
+            Action:
+              - s3:AbortMultipartUpload
+              - s3:GetBucketLocation
+              - s3:GetObject
+              - s3:ListBucket
+              - s3:ListBucketMultipartUploads
+              - s3:ListMultipartUploadParts
+              - s3:PutObject
+            Resource: arn:aws:s3:::panther-bootstrap-processeddata-*
+          - Effect: Allow
+            Action:
+              - s3:AbortMultipartUpload
+              - s3:GetBucketLocation
+              - s3:GetObject
+              - s3:ListBucket
+              - s3:ListBucketMultipartUploads
+              - s3:ListMultipartUploadParts
+              - s3:PutObject
+            Resource: arn:aws:s3:::panther-bootstrap-historicaldata-*
```

Since you have the same set of permissions for each bucket, you can combine these 3 statements into one statement with multiple resources:

```suggestion
          - Effect: Allow
            Action:
              - s3:AbortMultipartUpload
              - s3:GetBucketLocation
              - s3:GetObject
              - s3:ListBucket
              - s3:ListBucketMultipartUploads
              - s3:ListMultipartUploadParts
              - s3:PutObject
            Resource:
              - arn:aws:s3:::panther-bootstrap-processeddata-*
              - arn:aws:s3:::panther-bootstrap-historicaldata-*
              - arn:aws:s3:::panther-bootstrap-athenaresults-*
```

(with the caveat about bucket names I mentioned above)